application/models/services/QuestionGroupService.php

```diff
@@ -0,0 +1,265 @@
+            ], false);
+            $storeValid = $storeValid && $i10N->save();
+        }
+
+        return $storeValid;
+    }
+
+    /**
+     * Method to store and filter questionGroupData for editing a questionGroup
+     *
+     * @param QuestionGroup $oQuestionGroup
+     * @param array $aQuestionGroupData
+     *
+     * @return QuestionGroup
+     *
+     * @throws CException
+     */
+    public function editQuestionGroup($oQuestionGroup, $aQuestionGroupData)
+    {
+        $oQuestionGroup->setAttributes($aQuestionGroupData, false);
+        if ($oQuestionGroup == null) {
+            throw new CException("Object update failed, input array malformed or invalid");
+        }
+
+        $saved = $oQuestionGroup->save();
+        if (!$saved) {
+            throw new CException(
+                "Saved question group failed, object may be inconsistent"
+            );
+        }
+        return $oQuestionGroup;
+    }
+
+    /**
+     * Method to store and filter questionData for a new question
+     *
+     * @param int $surveyId
+     * @param array $aQuestionGroupData
+     *
+     * @return QuestionGroup
+     * @throws CException
+     */
+    private function newQuestionGroup($surveyId, $aQuestionGroupData = null)
+    {
+        $survey = $this->getSurvey($surveyId);
+        $aQuestionGroupData = array_merge([
+            'sid' => $survey->sid,
+        ], $aQuestionGroupData);
+        unset($aQuestionGroupData['gid']);
+
+        $oQuestionGroup = new QuestionGroup();
+        $oQuestionGroup->setAttributes($aQuestionGroupData, false);
+
+        if ($oQuestionGroup == null) {
+            throw new CException("Object creation failed, input array malformed or invalid");
+        }
+        // Always add at the end
+        $oQuestionGroup->group_order = safecount($survey->groups) + 1;
+        $saved = $oQuestionGroup->save();
+        if ($saved == false) {
+            throw new CException(
+                "Object creation failed, couldn't save.\n ERRORS:"
+                . print_r($oQuestionGroup->getErrors(), true)
+            );
+        }
+
+        $i10N = [];
+        foreach ($survey->allLanguages as $sLanguage) {
+            $i10N[$sLanguage] = new QuestionGroupL10n();
+            $i10N[$sLanguage]->setAttributes([
+                'gid'         => $oQuestionGroup->gid,
+                'language'    => $sLanguage,
+                'group_name'  => '',
+                'description' => '',
+            ], false);
+            $i10N[$sLanguage]->save();
+        }
+
+        return $oQuestionGroup;
+    }
+
+    /**
+     * @param int $surveyId
+     * @return QuestionGroup
+     */
+    public function getGroupData($surveyId)
+    {
+        $survey = $this->getSurvey($surveyId);
+
+        $questionGroup = new QuestionGroup('search');
+
+        if (isset($_GET['QuestionGroup']['group_name'])) {
+            $questionGroup->group_name = $_GET['QuestionGroup']['group_name'];
```

Comment:
Globals such as `_GET` should not be accessed, better to inject the request object.